site/content/blog/simplifying-package-authoring.md

```diff
@@ -0,0 +1,301 @@
+
+
+* **kctrl pkg init**: To initialize the Package and create the PackageInstall.
+
+* **kctrl dev**: To deploy PackageInstall CR locally.
+
+* **kctrl pkg release**: Create and upload the imgpkg bundle. Also, create the package to be released.
+
+* **kctrl pkg repo release**: Create the repo bundle so that it can be consumed by package consumers.
+
+Lets, try to create a `Dynatrace Operator` package by using the above commands.
+
+## Create a package
+
+### Prerequisites
+
+1. Install Carvel tools - [imgpkg](https://carvel.dev/imgpkg/docs/latest/install/), [kapp](https://carvel.dev/kapp/docs/latest/install/), [kbld](https://carvel.dev/kbld/docs/latest/install/), [vendir](https://carvel.dev/vendir/docs/latest/install/), [ytt](https://carvel.dev/ytt/docs/latest/install/).
+2. Identify K8s manifest which needs to be packaged. [`Dynatrace Operator`](https://github.com/Dynatrace/dynatrace-operator) releases the [`kubernetes.yaml`](https://github.com/Dynatrace/dynatrace-operator/releases) which can be packaged and be available for distribution.
+3. K8s cluster (I will be using minikube).
+4. OCI registry where the package bundle and repository bundles will be pushed (I will be using my DockerHub account).
+
+### kctrl pkg init
+
+```bash
+$ mkdir dynatrace 
+$ cd dynatrace
+$ kctrl pkg init
+```
+
+This command asks a few basic questions regarding how we want to initialize our package. Lets go through the question together:
+
+![Simplifying package authoring - pkg init basic details](/images/blog/simplifying-package-authoring-package-basic-details.png)
+
+* In this question, we need to enter the package name which will be a valid DNS subdomain name.
+
+![Simplifying package authoring - pkg init content option](/images/blog/simplifying-package-authoring-package-content-option.png)
+
+* Here, we need to enter from where to get the K8s manifest which needs to be packages. As mentioned earlier, `dynatrace` releases `kubernetes.yaml` as part of their github release artifacts. Hence, we will select `Option 2`.
+
+![Simplifying package authoring - pkg init dynatrace repository details](/images/blog/simplifying-package-authoring-dynatrace-repository-details.png)
+
+* In this few questions are being asked to enter github slug, tag/release version and the manifest file names etc.
+
+![Simplifying package authoring - pkg init vendir sync](/images/blog/simplifying-package-authoring-vendir-sync.png)
+
+* After entering all the above details, tool will download all the mentioned manifest locally and create two files - package-build.yml and pacakge-resources.yml. `Package-resources.yml` file contain Package, PackageInstall and PackageMetadata while `package-build.yml` contains PackageBuild. More information on PackageBuild can be found in this [link](). All the values entered above have been recorded in these files. These files will be used by the `dev` and `pkg release` command subsequently. 
```

Comment:
"After entering all the above details, **tool will download**"

What tool? Vendir? If it's vendir then this should be changed to "After entering all the above details, vendir will download"

Reply:
Adopted. Changed to 

"After entering all the above details, [`vendir`](https://carvel.dev/vendir/) will download all the mentioned manifest locally. `kctrl` will generate two files - package-build.yml and pacakge-resources.yml as output. `Package-resources.yml` file..."